examples/src/main/scala/org/apache/spark/examples/ml/BisectingKMeansExample.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.examples.ml
+
+// scalastyle:off println
+
+// $example on$
+import org.apache.spark.ml.clustering.BisectingKMeans
+import org.apache.spark.mllib.linalg.{Vectors, VectorUDT}
+import org.apache.spark.sql.{DataFrame, Row, SparkSession}
+import org.apache.spark.sql.types.{StructField, StructType}
+// $example off$
+
+/**
+ * An example demonstrating a bisecting k-means clustering.
+ * Run with
+ * {{{
+ * bin/run-example ml.BisectingKMeansExample
+ * }}}
+ */
+object BisectingKMeansExample {
+
+  def main(args: Array[String]): Unit = {
+    // Creates a SparkSession
+    val spark = SparkSession.builder.appName("BisectingKMeansExample").getOrCreate()
+
+    // $example on$
+    // Crates a DataFrame
+    val rowRDD = spark.read.text("data/mllib/kmeans_data.txt").rdd
```

Comment:
So, in updating the Kmeans examples, I had the same issues that make this code a bit ugly. I came up with:

```scala
val vecAssembler = new VectorAssembler()
      .setInputCols(Array("x", "y", "z"))
      .setOutputCol("features")

    val schema = StructType(Array(
      StructField("x", DataTypes.DoubleType),
      StructField("y", DataTypes.DoubleType),
      StructField("z", DataTypes.DoubleType)))

    val dataset = vecAssembler.transform(
      spark.read
      .format("csv")
      .option("sep", " ")
      .schema(schema)
      .load("data/mllib/kmeans_data.txt"))
```

I think it's a little bit better since we don't convert to RDD in what we claim is the "dataframe API," but I am not certain what is best. Thoughts? @MLnick